### PR TITLE
Implementation for `get_search_suggestions`

### PR DESCRIPTION
**Pull request by omgupta15**
Please refer to #370 for the description.

### CODE REVIEW

File: ytmusicapi/mixins/browsing.py
Comment (sigma67) [inline ytmusicapi/mixins/browsing.py:1]: What's the reason for this change?

File: ytmusicapi/mixins/search.py [hunk 1]
```diff
@@ -203,12 +203,100 @@ def search(self,
                                      category))
 
             if filter:  # if filter is set, there are continuations
-                request_func = lambda additionalParams: self._send_request(
-                    endpoint, body, additionalParams)
-                parse_func = lambda contents: parse_search_results(contents, type, category)
+
```
Comment (sigma67) [inline ytmusicapi/mixins/search.py:206]: What's the reason for this change?

File: ytmusicapi/mixins/search.py [hunk 2]
```diff
@@ -203,12 +203,100 @@ def search(self,
...
+            usually bold the text typed by the user).
+            Default: False, returns the list of search suggestions in plain text.
+        :return: List of search suggestion results depending on detailed_runs param.
+            Examples:
+                1.
+                Params:
+                    query = 'fade'
+                    detailed_runs = False
+
+                Response:
```
Comment (sigma67) [inline ytmusicapi/mixins/search.py:236]: I think you need to format this differently so it gets formatted correctly as JSON in the built documentation. Please check the other docstrings and build the documentation (run `./make.bat html` on Windows or `make html` on Linux from the docs folder).

**Review by sigma67 — CHANGES_REQUESTED** [review]
Looks almost ready. Some questions and a docstring change.

File: ytmusicapi/mixins/browsing.py
Comment (omgupta15) [reply ytmusicapi/mixins/browsing.py:1]: This change was made automatically when I used the command `pre-commit run`.

File: ytmusicapi/mixins/search.py (hunk 1 above)
Comment (omgupta15) [reply ytmusicapi/mixins/search.py:206]: This change was made automatically when I used the command `pre-commit run`.

File: ytmusicapi/mixins/browsing.py
Comment (sigma67) [reply ytmusicapi/mixins/browsing.py:1]: Please revert this one (mixings/browsing.py) as there are no other changes to this file anymore

File: ytmusicapi/mixins/search.py (hunk 1 above)
Comment (sigma67) [reply ytmusicapi/mixins/search.py:206]: In that case it's fine.

**Review by sigma67 — CHANGES_REQUESTED** [review]
Docstring looks good now, revert mixings/browsing.py and we're good to go

**Review by sigma67 — APPROVED** [review]
